polyfills/Object.getOwnPropertyNames/polyfill.js

```diff
@@ -1,12 +1,27 @@
 Object.getOwnPropertyNames = function getOwnPropertyNames(object) {
-	if (object !== Object(object)) {
-		throw new TypeError('Object.getOwnPropertyNames called on non-object');
+	var buffer = [];
+	var key;
+
+	// Non-enumerable properties cannot be discovered but can be checked for by name.
+	// Define those used internally by JS to allow an incomplete solution
+	var commonProps = ['length', "name", "arguments", "caller", "prototype", "observe", "unobserve"];
+
+	if (typeof object === 'undefined' || object === null) {
+		throw new TypeError('Cannot convert undefined or null to object');
 	}
 
-	var buffer = [], key;
+	object = Object(object);
 
+	// Enumerable properties only
 	for (key in object) {
-		buffer.push(key);
+		if (object.hasOwnProperty(key)) {
```

Comment:
Have you tested this on IE8?  

The implementation found in this commit (3385d9728b42f3fd9302fa6511634a591babc0ad) works for ie8. Do you need a PR for this?